src/main/java/com/nike/cerberus/service/CertificateService.java

```diff
@@ -274,11 +268,24 @@ public void generateCerts(File certDir,
             Session session = new Session(acmeServerUrl, userKeyPair);
             Registration registration = findOrRegisterAccount(session, contactEmail);
 
+            Map<String, Collection<Challenge>> domainChallengeCollectionMap = new HashMap<>();
             for (String name : names) {
                 Authorization authorization = registration.authorizeDomain(name);
-                getChallenges(authorization).forEach(challenge -> doChallenge(challenge, name, hostedZoneId));
+                domainChallengeCollectionMap.put(name, getChallenges(authorization));
             }
 
+            ThreadPoolExecutor challengeExecutorService = (ThreadPoolExecutor) Executors.newFixedThreadPool(names.size());
+            domainChallengeCollectionMap.forEach((name, challengeCollection) -> {
+                challengeExecutorService.execute(() ->
+                        challengeCollection.forEach(challenge -> doChallenge(challenge, name, hostedZoneId)));
+            });
+
+            do {
+                log.info("Waiting for all challenges to complete before continuing");
```

Comment:
You might log the active count in your loop

Reply:
will add this.